internal/commands/accesslist/delete.go

```diff
@@ -0,0 +1,195 @@
+	HelpText: `Removes an existing entry from the Access List of your Realm app. You will be
+prompted to select an IP address or CIDR block if none is provided in the
+initial command.`,
+}
+
+// CommandDelete is the `accesslist delete` command
+type CommandDelete struct {
+	inputs deleteInputs
+}
+
+// Flags is the command flags
+func (cmd *CommandDelete) Flags() []flags.Flag {
+	return []flags.Flag{
+		cli.AppFlagWithContext(&cmd.inputs.App, "to remove an entry in its Access List"),
+		cli.ProjectFlag(&cmd.inputs.Project),
+		cli.ProductFlag(&cmd.inputs.Products),
+		flags.StringSliceFlag{
+			Value: &cmd.inputs.Addresses,
+			Meta: flags.Meta{
+				Name: "ip",
+				Usage: flags.Usage{
+					Description: "Specify the IP address or CIDR block that you would like to delete",
+				},
+			},
+		},
+	}
+}
+
+// Inputs is the command inputs
+func (cmd *CommandDelete) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Handler is the command handler
+func (cmd *CommandDelete) Handler(profile *user.Profile, ui terminal.UI, clients cli.Clients) error {
+	app, err := cli.ResolveApp(ui, clients.Realm, cmd.inputs.Filter())
+	if err != nil {
+		return err
+	}
+
+	allowedIPs, err := clients.Realm.AllowedIPs(app.GroupID, app.ID)
+	if err != nil {
+		return err
+	}
+
+	selected, err := cmd.inputs.resolveAllowedIP(ui, allowedIPs)
+	if err != nil {
+		return err
+	}
+
+	if len(selected) == 0 {
+		ui.Print(terminal.NewTextLog("No IP addresses or CIDR blocks to delete"))
+		return nil
+	}
+
+	outputs := make(deleteAllowedIPOutputs, len(selected))
+	for i, allowedIP := range selected {
+		err := clients.Realm.AllowedIPDelete(app.GroupID, app.ID, allowedIP.ID)
+		outputs[i] = deleteAllowedIPOutput{allowedIP, err}
+	}
+
+	sort.SliceStable(outputs, func(i, j int) bool {
+		return outputs[i].err != nil && outputs[j].err == nil
+	})
+
+	tableRows := make([]map[string]interface{}, 0, len(outputs))
+	for _, output := range outputs {
+		row := map[string]interface{}{
+			headerAddress: output.allowedIP.Address,
+			headerComment: output.allowedIP.Comment,
+		}
+		tableRows = append(tableRows, row)
+		tableRowDelete(output, row)
+	}
+
+	ui.Print(terminal.NewTableLog(
+		fmt.Sprintf("Deleted %d IP address(es) and CIDR block(s)", len(outputs)),
+		deleteTableHeaders,
+		tableRows...,
+	))
+
+	return nil
+}
+
+func (i *deleteInputs) Resolve(profile *user.Profile, ui terminal.UI) error {
+	if err := i.ProjectInputs.Resolve(ui, profile.WorkingDirectory, false); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (i *deleteInputs) resolveAllowedIP(ui terminal.UI, allowedIPs []realm.AllowedIP) ([]realm.AllowedIP, error) {
+	if len(allowedIPs) == 0 {
+		return nil, nil
+	}
+
+	if len(i.Addresses) > 0 {
+		allowedIPsByAddress := make(map[string]realm.AllowedIP, len(allowedIPs))
+		for _, allowedIP := range allowedIPs {
+			allowedIPsByAddress[allowedIP.Address] = allowedIP
+		}
+
+		allowedIPs := make([]realm.AllowedIP, 0, len(i.Addresses))
```

Comment:
do we want to account for warning the user for any addresses they may have specified that could not be found?  would probably need to check with the project team if that's worth just a WARNING or true ERROR

Reply:
@christineschen

Reply:
talked to Christine and Sumedha, we're going to go with error-ing out if they provide us with an address that doesn't exist in their access list, whether they provide a single address or multiple

Reply:
sounds good to me... just want to call out that the error should still probably indicate _all_ of the addresses that aren't found, rather than simply error out on the first one